services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/QueryEnhancedRequest.java

```diff
@@ -192,7 +206,8 @@ public int hashCode() {
         private Integer limit;
         private Boolean consistentRead;
         private Expression filterExpression;
-        private List<String> attributesToProject;
+        private List<String> attributesToProjectStringList;
```

Comment:
attributesToProjectStringList attribute is required to maintain the order of the older attributeToProject in cases of resets.

Reply:
Is the order important? Conceptually I think of this as a 'set'.

Reply:
`QueryEnhancedRequest.builder(). attributesToProject("One", "two ", "three"). addNestedAttributesToProject("foo", "bar").attributesToProject("bazz");`
  The expected output for this should be
getAttributesToProject --> [foo.bar, bazz]
Thus I have kept a Temp List on the Builder called attributesToProjectStringList where you had commented that order is not required. I will rename to attributesToProjectTempList;